src/wasm2js.h

```diff
@@ -410,9 +410,12 @@ Ref Wasm2JSBuilder::processWasm(Module* wasm, Name funcName) {
       ValueBuilder::makeName("// EMSCRIPTEN_START_FUNCS\n"));
   }
   // functions
-  ModuleUtils::iterDefinedFunctions(*wasm, [&](Function* func) {
-    asmFunc[3]->push_back(processFunction(wasm, func));
-  });
+  // Don't use iterDefinedFunctions because the iterator could be invalidated
```

Comment:
One of the functions called like five levels below this actually adds new utility functions to the module. This never caused problems before because previous optimizations removed functions, making room in the vector for the utility functions to be added without reallocating. I found this problem because a round trip I had here replaced the old vector with extra capacity with a smaller fresh vector.